### PR TITLE
issue-1070: Use testKube.CleanNamespaceAndLogOnPanic in all tests

### PR DESCRIPTION
**Pull request by durgeshanaokar**
Pull request for https://github.com/infinispan/infinispan-operator/issues/1070
~~~
Would be nice to embed testKube.CleanNamespaceAndLogOnPanic for all our tests (Core and Cross-site)
Now only Backup/Restore utilizes this function.
Probably this function could be adopted a bit
~~~

### CODE REVIEW

File: test/e2e/main/main_test.go [hunk 1]
```diff
@@ -83,6 +84,7 @@ func TestUpdateOperatorPassword(t *testing.T) {
 	spec.Name = name
 	// Register it
 	testKube.CreateInfinispan(spec, tutils.Namespace)
+	defer testKube.CleanNamespaceAndLogOnPanic(tutils.Namespace)
```
Comment (dmvolod) [inline test/e2e/main/main_test.go:87]: @durgeshanaokar it's required to replace DeleteInfinispan with this function
Also need to validate, is this work with the Batch entries to be removed and traced

**Review by dmvolod — CHANGES_REQUESTED** [review]

**Comment by dmvolod** [comment]
As main tests partially running in parallel, it's necessary to add `name` parameter to the `testKube.CleanNamespaceAndLogOnPanic` and delete by name if value is provided

File: test/e2e/main/main_test.go [hunk 2]
```diff
@@ -51,6 +51,7 @@ func TestOperatorUpgrade(t *testing.T) {
 	switch tutils.OperatorUpgradeStage {
 	case tutils.OperatorUpgradeStageFrom:
 		testKube.CreateInfinispan(spec, tutils.Namespace)
+		defer testKube.CleanNamespaceAndLogOnPanic(tutils.Namespace)
```
Comment (dmvolod) [inline test/e2e/main/main_test.go:54]: We should not do cleanup for the initial upgrade stage

File: test/e2e/main/main_test.go [hunk 3]
```diff
@@ -55,7 +55,8 @@ func TestOperatorUpgrade(t *testing.T) {
 		testKube.WaitForInfinispanCondition(spec.Name, spec.Namespace, ispnv1.ConditionWellFormed)
 	case tutils.OperatorUpgradeStageTo:
 		if tutils.CleanupInfinispan == "TRUE" {
-			defer testKube.DeleteInfinispan(spec, tutils.SinglePodTimeout)
+			defer testKube.CleanNamespaceAndLogOnPanic(tutils.Namespace, nil)
+			//defer testKube.DeleteInfinispan(spec, tutils.SinglePodTimeout)
```
Comment (ryanemerson) [inline test/e2e/main/main_test.go:59]: Thanks for the PR @durgeshanaokar. Can you remove all the commented lines as these no longer serve a purpose.

File: test/e2e/main/main_test.go [hunk 4]
```diff
@@ -83,7 +84,9 @@ func TestUpdateOperatorPassword(t *testing.T) {
 	spec.Name = name
 	// Register it
 	testKube.CreateInfinispan(spec, tutils.Namespace)
-	defer testKube.DeleteInfinispan(spec, tutils.SinglePodTimeout)
+	spec.Labels = map[string]string{"test-name": "TestUpdateOperatorPassword"}
```
Comment (dmvolod) [inline test/e2e/main/main_test.go:87]: @durgeshanaokar please pay attention, that `t.Name` is a test name and wan't to be hardcoded

File: test/e2e/utils/kubernetes.go [hunk 5]
```diff
@@ -168,11 +168,26 @@ func (k TestKubernetes) CleanNamespaceAndLogOnPanic(namespace string) {
 
 	if CleanupInfinispan == "TRUE" || panicVal == nil {
 		ctx := context.TODO()
-		ExpectMaybeNotFound(k.Kubernetes.Client.DeleteAllOf(ctx, &ispnv2.Batch{}, opts...))
+		if spacLabel == nil {
+			ExpectMaybeNotFound(k.Kubernetes.Client.DeleteAllOf(ctx, &ispnv2.Batch{}, opts...))
+			ExpectMaybeNotFound(k.Kubernetes.Client.DeleteAllOf(ctx, &ispnv2.Cache{}, opts...))
+			ExpectMaybeNotFound(k.Kubernetes.Client.DeleteAllOf(ctx, &ispnv1.Infinispan{}, opts...))
+		} else {
+			ispnBatch := &ispnv2.Batch{}
```
Comment (dmvolod) [inline test/e2e/utils/kubernetes.go:176]: Label should be added to the `opts` as `MatchingLabels`, not to the CR object.

File: test/e2e/main/main_test.go (hunk 4 above)
Comment (durgeshanaokar) [reply test/e2e/main/main_test.go]: Ack, will take care

File: test/e2e/utils/kubernetes.go [hunk 6]
```diff
@@ -168,11 +168,22 @@ func (k TestKubernetes) CleanNamespaceAndLogOnPanic(namespace string) {
 
 	if CleanupInfinispan == "TRUE" || panicVal == nil {
 		ctx := context.TODO()
-		ExpectMaybeNotFound(k.Kubernetes.Client.DeleteAllOf(ctx, &ispnv2.Batch{}, opts...))
+		if spacLabel == nil {
+			ExpectMaybeNotFound(k.Kubernetes.Client.DeleteAllOf(ctx, &ispnv2.Batch{}, opts...))
```
Comment (dmvolod) [inline test/e2e/utils/kubernetes.go:172]: You need to define Delete twice, you just need to define `opts` for different cases and keep delete function in the single place

File: test/e2e/main/main_test.go [hunk 7]
```diff
@@ -195,7 +200,9 @@ func TestNodeStartup(t *testing.T) {
 	defer os.Unsetenv(v1.OperatorPodTargetLabelsEnvVarName)
 	// Register it
 	testKube.CreateInfinispan(spec, tutils.Namespace)
-	defer testKube.DeleteInfinispan(spec, tutils.SinglePodTimeout)
+	spec.Labels = map[string]string{"test-name": t.Name()}
```
Comment (dmvolod) [inline test/e2e/main/main_test.go:203]: Label should be defined before Infinispan create as `spec.Labels["test-name"] = t.Name()` as labels already utilized in this test

File: test/e2e/utils/kubernetes.go [hunk 8]
```diff
@@ -149,7 +149,7 @@ func (k TestKubernetes) NewNamespace(namespace string) {
 	ExpectNoError(err)
 }
 
-func (k TestKubernetes) CleanNamespaceAndLogOnPanic(namespace string) {
+func (k TestKubernetes) CleanNamespaceAndLogOnPanic(namespace string, spacLabel map[string]string) {
```
Comment (dmvolod) [inline test/e2e/utils/kubernetes.go:152]: Misprint in `spacLabel`

File: test/e2e/main/main_test.go [hunk 9]
```diff
@@ -83,7 +84,9 @@ func TestUpdateOperatorPassword(t *testing.T) {
 	spec.Name = name
 	// Register it
 	testKube.CreateInfinispan(spec, tutils.Namespace)
-	defer testKube.DeleteInfinispan(spec, tutils.SinglePodTimeout)
+	spec.Labels = map[string]string{"test-name": t.Name()}
```
Comment (dmvolod) [inline test/e2e/main/main_test.go:87]: Also labels assigned after Infinispan CR creation.
@durgeshanaokar please validate all possible places for this issue.

**Pull request closed without merging**